features/facebookImport/src/model/storage.js

```diff
@@ -51,34 +70,34 @@ export class ZipFile {
     }
 
     async _ensureCachedEntries() {
-        if (this._entriesSet !== null) return;
+        if (this._entriesPathHash !== null) return;
         const entriesList = await this._readEntriesList();
-        const entriesIdList = entriesList.map((entry) => entry.id);
-        this._entriesSet = new Set(entriesIdList);
+        this._entriesPathHash = new Map();
+        entriesList.forEach((entry) =>
+            this._entriesPathHash.set(
+                entry.path,
+                new ZipFileEntry(this._pod, this, entry.id, entry.path)
+            )
+        );
     }
 
-    async getEntries() {
+    async hasFilePath(entryPath) {
         await this._ensureCachedEntries();
-        return [...this._entriesSet];
+        return this._entriesPathHash.has(entryPath);
     }
 
-    async hasEntry(entryId) {
+    async fileEntryForPath(entryPath) {
         await this._ensureCachedEntries();
-        return this._entriesSet.has(entryId);
+        return this._entriesPathHash.get(entryPath);
     }
 
-    async data() {
-        return this.getContent(this.id);
-    }
-
-    async stat(entry) {
-        const { polyOut } = this._pod;
-        return polyOut.stat(entry);
+    async getEntries() {
+        await this._ensureCachedEntries();
+        return [...this._entriesPathHash.values()];
     }
 
-    async getContent(entry) {
-        const { polyOut } = this._pod;
-        return polyOut.readFile(entry);
+    async data() {
```

Comment:
Maybe this could be eliminated altogether?

Reply:
yep don't think it is used anywhere else